external/fv3fit/tests/data_/test_netcdf.py

```diff
@@ -32,3 +36,75 @@ def test_NCDirLoader(tmp_path: Path):
 
 def test_Netcdf_from_dict():
     fv3fit.data.NCDirLoader.from_dict({"url": "some/path"})
+
+
+NX, NY, NZ, NT = 5, 6, 8, 10
+
+
+@contextlib.contextmanager
+def temporary_netcdfs_dir():
+    dir = tempfile.TemporaryDirectory()
+
+    # Test datasets that have different number of timesteps
+    ds = xr.Dataset(
+        data_vars={
+            "a_sfc": xr.DataArray(np.random.randn(NX, NY, NT), dims=["x", "y", "time"]),
+            "b": xr.DataArray(
+                np.random.randn(NX, NY, NZ, NT), dims=["x", "y", "z", "time"]
+            ),
+        }
+    )
+
+    ds.to_netcdf(f"{dir.name}/0.nc")
+    ds.isel(time=slice(0, 2)).to_netcdf(f"{dir.name}/1.nc")
+    yield dir.name
+
+
+@pytest.mark.parametrize(
+    "dim_order, expected_shape_first_batch",
+    [
+        (["time", "x", "y", "z"], (NT, NX, NY, NZ)),
+        (["time", "y", "z", "x"], (NT, NY, NZ, NX)),
+    ],
+)
+def test_NCDirLoader_dim_order(dim_order, expected_shape_first_batch):
+    with temporary_netcdfs_dir() as data_path:
+        loader = fv3fit.data.NCDirLoader(
+            url=data_path, dim_order=dim_order, shuffle=False, varying_first_dim=True,
+        )
+        dataset = loader.open_tfdataset(
+            local_download_path=None, variable_names=["a_sfc", "b"]
+        )
+        first_batch = next(iter(dataset))
+        assert first_batch["b"].shape == expected_shape_first_batch
+
+
+def test_NCDirLoader_varying_first_dim_allowed():
```

Comment:
optional: consider adding a test for when varying_first_dim is not allowed but the first dimension does vary.